### ISSUE TITLE
zalenium cannot get access to the host via mac.host.local

### ISSUE THREAD

**Issue by xprommer** [zalando/zalenium#622]
Zalenium Image Version(s):   3.12.0d

Docker Version:  17.06.1-ce

OS: Mac OS X 10.10.5

Docker Command to start Zalenium:  
```bash
docker run -d --name zalenium -p 4444:4444 \
        -v /var/run/docker.sock:/var/run/docker.sock \
        -v /tmp/videos:/home/seluser/videos \
        --privileged dosel/zalenium start
```

## Expected Behavior -
according to zalenium's docs `mac.host.local` should be accessible from zalenium

## Actual Behavior -
I'm running a node server on 8080 port on host machine, and zalenium cannot resolve `mac.host.local`:
```
curl http://mac.host.local:8080  => curl: (6) Could not resolve host: mac.host.local
```
but when I'm trying via IP it works:
```
curl http://10.166.53.19:8080 => <!doctype html><html>....</html>
```

**Comment by pearj** [comment]
Hi @xprommer I can't find any references to ".local" in https://opensource.zalando.com/zalenium/ or https://github.com/zalando/zalenium/blob/master/README.md what makes you think that it is supported?

I don't think it'd be that easy to get going, as we'd need to bundle avahi-daemon (which provides mDNS) in elgalu/docker-selenium.  It's something that we'd want to turned off by default, as you wouldn't want mDNS operating inside a Kubernetes cluster for example.

The other approach you can take is to run zalenium with `--net=host` and then you can simply use `localhost:8080` and not need to worry about `mac.host.local` at all.

Eg:

```
docker run -d --name zalenium -p 4444:4444 \
        --net=host \
        -v /var/run/docker.sock:/var/run/docker.sock \
        -v /tmp/videos:/home/seluser/videos \
        --privileged dosel/zalenium start
```

**Comment by xprommer** [comment]
Hi @pearj you can find this on https://opensource.zalando.com/zalenium/ in `Special Cases and Customisations ` section if you open `Accessing the host` item.  Regarding `--net=host` it doesn't work on mac, that's why zalenium team introduced `mac.host.local` hostname and unfortunately this doesn't work as well. So in order to make it work I use `--add-host` option with my host and ip

**Comment by pearj** [comment]
Fair enough, @diemol should know when that option worked last.

**Comment by diemol** [comment]
Hi @xprommer,

Some time ago, the way to access the host from Docker under Mac and Windows was kind of hidden in the Docker docs, so we tried to make it more transparent and added support for it in Zalenium.

Nowadays, it is better documented, as it can be seen here:
https://docs.docker.com/docker-for-mac/networking/#use-cases-and-workarounds
https://docs.docker.com/docker-for-windows/networking/#use-cases-and-workarounds

So please use `gateway.docker.internal` from now on (keep in mind that you need a recent version of Docker).

I updated the Zalenium docs and also removed the code we used to support that.

**Comment by mcopjan** [comment]
Hi @diemol. I am testing web site which is company local domain and  I want to access this page from inside zalenium containers. Even after reading documentation I am not really sure how to get to the page as my containers are not running on the the same box where tested web site is hosted so I cannot use e.g. "http://gateway.docker.internal:8080". Is there a way how I can "share" containers with company DNS and see my test pages from inside Zalenium containers? Thanks

**Comment by mcopjan** [comment]
Alright so the problem above was caused by me not knowing much about Docker. I was originally facing issues when pulling Zalenium images from public repo. Some posts suggested to switch DNS settings in Docker daemon from automatic to fixed (8.8.8.8), which I did and pulled the images without issues. I forgot about this change unfortunately therefore my docker containers were not using our local DNS, switching back to automatic resolved the issues.
